fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/FoldConstantRuleOnFE.java

```diff
@@ -287,23 +295,31 @@ public Expression visitCast(Cast cast, ExpressionRewriteContext context) {
 
     @Override
     public Expression visitBoundFunction(BoundFunction boundFunction, ExpressionRewriteContext context) {
+        boundFunction = rewriteChildren(boundFunction, context);
         //functions, like current_date, do not have arg
         if (boundFunction.getArguments().isEmpty()) {
             return boundFunction;
         }
-        if (!ExpressionUtils.isAllLiteral(boundFunction.getArguments())) {
-            return boundFunction;
+        Optional<Expression> checkedExpr = checkNeedCalculate(boundFunction);
+        if (checkedExpr.isPresent()) {
+            return checkedExpr.get();
         }
         return ExpressionEvaluator.INSTANCE.eval(boundFunction);
     }
 
     @Override
     public Expression visitBinaryArithmetic(BinaryArithmetic binaryArithmetic, ExpressionRewriteContext context) {
+        binaryArithmetic = rewriteChildren(binaryArithmetic, context);
+        Optional<Expression> checkedExpr = checkNeedCalculate(binaryArithmetic);
```

Comment:
i think we cannot do fold constant if this function return Optional.empty(),right?

Reply:
no, the checkNeedCalculate() means that if the binary arithmetic is no need to be fold, we return an option contain the expression and return, otherwise we fold the expression